apis/v1alpha1/opampbridge_webhook_test.go

```diff
@@ -0,0 +1,197 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestOpAMPBridgeDefaultingWebhook(t *testing.T) {
+	one := int32(1)
+	five := int32(5)
+
+	tests := []struct {
+		name        string
+		opampBridge OpAMPBridge
+		expected    OpAMPBridge
+	}{
+		{
+			name:        "provide only required values in spec",
+			opampBridge: OpAMPBridge{},
+			expected: OpAMPBridge{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app.kubernetes.io/managed-by": "opentelemetry-operator",
+					},
+				},
+				Spec: OpAMPBridgeSpec{
+					Replicas:        &one,
+					UpgradeStrategy: UpgradeStrategyAutomatic,
+				},
+			},
+		},
+		{
+			name: "provided optional values in spec",
+			opampBridge: OpAMPBridge{
+				Spec: OpAMPBridgeSpec{
+					Replicas:        &five,
+					UpgradeStrategy: "adhoc",
+				},
+			},
+			expected: OpAMPBridge{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app.kubernetes.io/managed-by": "opentelemetry-operator",
+					},
+				},
+				Spec: OpAMPBridgeSpec{
+					Replicas:        &five,
+					UpgradeStrategy: "adhoc",
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			test.opampBridge.Default()
+			assert.Equal(t, test.expected, test.opampBridge)
+		})
+	}
+}
+
+// TODO: a lot of these tests use .Spec.MaxReplicas and .Spec.MinReplicas. These fields are
+// deprecated and moved to .Spec.Autoscaler. Fine to use these fields to test that old CRD is
+// still supported but should eventually be updated.
```

Comment:
Do we need this todo?

Reply:
Can you add a test for the max replicas = 1?

Reply:
> Do we need this todo?

Removed.